.github/workflows/land-blocking.yml

```diff
@@ -11,19 +11,72 @@ jobs:
     name: Build images and run cluster test
     runs-on: self-hosted
     steps:
-    - uses: actions/checkout@v1
-    - name: Setup env
-      run: |
-        echo "::set-env name=LIBRA_GIT_REV::$(git rev-parse --short=8 HEAD)"
-    - name: Build, tag and push images
-      run: |
-        docker/build-aws.sh --build-all --version $LIBRA_GIT_REV --addl_tags canary
-    - name: Launch cluster test
-      run: |
-        JUMPHOST=${{secrets.CLUSTER_TEST_JUMPHOST}} \
-        ./scripts/cti \
-          --marker land \
-          --tag dev_$LIBRA_GIT_REV \
-          --report report.json \
-          --run bench \
-          -- --duration 60
+      - uses: actions/checkout@v1
+      - name: Setup env
+        run: |
+          echo "::set-env name=LIBRA_GIT_REV::$(git rev-parse --short=8 HEAD)"
+      - name: Build, tag and push images
+        run: |
+          docker/build-aws.sh --build-all --version $LIBRA_GIT_REV --addl_tags canary
+      - name: Launch cluster test
+        run: |
+          JUMPHOST=${{secrets.CLUSTER_TEST_JUMPHOST}} \
+          ./scripts/cti \
+            --marker land \
+            --tag dev_$LIBRA_GIT_REV \
+            --report report.json \
+            --run bench \
+            -- --duration 60
+      - name: Post test results on PR
+        if: always()
+        uses: actions/github-script@0.4.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            should_fail = false;
+            const fs = require('fs');
+            fs.readFile('report.json', 'utf-8', (err, data) => {
```

Comment:
Please use `await` here. Since Node 11 fs.readFile returns promises, and the script action uses Node 12. See https://stackoverflow.com/a/56821924. That will let you get rid of the `should_fail` stuff too.

Reply:
The `should_fail` is for separate failure cases (coming in the next few diffs)
- cluster test fails
- cluster test passes but TPS regresses x %
- cluster test passes but latency regresses y %
etc

Reply:
Updated but kept should_fail, as we will need it for the next phase (fail the workflow if there is perf regression).